theano/scan_module/scan_checkpoints.py

```diff
@@ -96,16 +101,29 @@ def scan_checkpoints(fn, sequences=[], outputs_info=None, non_sequences=[],
     if n_steps is None:
         n_steps = sequences[0].shape[0]
 
-    # Compute the number of steps of the inner and of the outer scan
-    o_n_steps = theano.tensor.cast(n_steps / save_every_N, 'int64')
-    i_n_steps = save_every_N
+    # Compute the number of steps of the outer scan
+    o_n_steps = theano.tensor.cast(theano.tensor.ceil(n_steps / save_every_N),
+                                   'int64')
+    i_n_steps = save_every_N * theano.tensor.ones((o_n_steps,), 'int64')
+    mod = n_steps % save_every_N
+    sign = theano.tensor.sgn(mod)
```

Comment:
@nouiz So this is not to support `n_steps` negative. This is to compute how many steps the last inner scan must run.

And for the negative steps, it is in the docstring of `scan.py`